datatype.go

```diff
@@ -32,27 +32,45 @@ const (
 	unSupportedType
 )
 
-var snowflakeTypes = [...]string{"FIXED", "REAL", "TEXT", "DATE", "VARIANT",
-	"TIMESTAMP_LTZ", "TIMESTAMP_NTZ", "TIMESTAMP_TZ", "OBJECT", "ARRAY",
-	"BINARY", "TIME", "BOOLEAN", "NULL", "SLICE", "CHANGE_TYPE", "NOT_SUPPORTED"}
+var stringToSnowflakeType = map[string]snowflakeType{
+	"FIXED":         fixedType,
+	"REAL":          realType,
+	"TEXT":          textType,
+	"DATE":          dateType,
+	"VARIANT":       variantType,
+	"TIMESTAMP_LTZ": timestampLtzType,
+	"TIMESTAMP_NTZ": timestampNtzType,
+	"TIMESTAMP_TZ":  timestampTzType,
+	"OBJECT":        objectType,
+	"ARRAY":         arrayType,
+	"BINARY":        binaryType,
+	"TIME":          timeType,
+	"BOOLEAN":       booleanType,
+	"NULL":          nullType,
+	"SLICE":         sliceType,
+	"CHANGE_TYPE":   changeType,
+	"NOT_SUPPORTED": unSupportedType}
 
-func (st snowflakeType) String() string {
-	return snowflakeTypes[st]
+var snowflakeTypeToString = invertMap(stringToSnowflakeType)
+
+func invertMap(m map[string]snowflakeType) map[snowflakeType]string {
```

Comment:
I wonder what would happen if some duplication occurred here... Is it only used for `String` method? Is it safe?

Reply:
Added check for duplicated values